aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadOptions.java

```diff
@@ -0,0 +1,124 @@
+package com.amazonaws.mobileconnectors.s3.transferutility;
+
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+
+/**
+ * A class that can hold all of the optional parameters (bucket, metadata, cannedAcl, transferListener).
+ */
+public class UploadOptions {
```

Comment:
should probably add `final` keyword to this too, since we don't want people to extend this class.